plugins/stack-overflow-backend/src/search/StackOverflowQuestionsCollatorFactory.ts

```diff
@@ -98,12 +102,17 @@ export class StackOverflowQuestionsCollatorFactory
       config.getOptionalString('stackoverflow.baseUrl') ||
       'https://api.stackexchange.com/2.3';
     const maxPage = options.maxPage || 100;
+    const requestParams =
+      config.getOptional<StackOverflowQuestionsRequestParams>(
+        'stackoverflow.requestParams',
+      );
```

Comment:
```suggestion
    const requestParams =
      config.getOptionalConfig('stackoverflow.requestParams')?.get<StackOverflowQuestionsRequestParams>();
```

this has the additional benefit of ensuring that the thing you're getting back is an object, instead of for example a string.

Reply:
TIL, thanks for the suggestion 🙌🏻 : https://github.com/backstage/backstage/pull/20456/files#diff-b6834ddfd3fadd7101daffc2464423f035c153275ba72e7423e5388e1e35dd61R108